src/main/java/com/google/sps/servlets/SignUpWithGoogleServlet.java

```diff
@@ -0,0 +1,42 @@
+package com.google.sps.servlets;
+
+import com.google.cloud.datastore.Datastore;
+import com.google.cloud.datastore.DatastoreOptions;
+import com.google.cloud.datastore.Entity;
+import com.google.cloud.datastore.FullEntity;
+import com.google.cloud.datastore.KeyFactory;
+import com.google.cloud.datastore.Key;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.sql.rowset.WebRowSet;
+import org.jsoup.Jsoup;
+import org.jsoup.safety.Whitelist;
+
+@WebServlet("/signup")
+public class SignUpWithGoogleServlet extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    String nameValue = Jsoup.clean(request.getParameter("name"), Whitelist.none());
+    String emailValue = Jsoup.clean(request.getParameter("email"), Whitelist.none());
+    String idValue = Jsoup.clean(request.getParameter("ID"),Whitelist.none());
+    System.out.println("hello!"+nameValue);
+
+    Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
+    KeyFactory keyFactory = datastore.newKeyFactory().setKind("User");
+    FullEntity newUser =
+        Entity.newBuilder(keyFactory.newKey())
+            .set("name", nameValue)
+            .set("email", emailValue)
+            .set("ID",idValue)
+            .build();
```

Comment:
This is good for this PR.

An idea for the next PR where we resolve the issue with duplicate rows:

To avoid the duplicate row issue, we should only create the newUser entity if we know it doesn't exit. We can check if it exists if we know the key. Therefore, we should change the key to the idValue, since that's something that we are given in the request and we also know it should be unique for each user.

Something like the example in this one: https://cloud.google.com/appengine/docs/standard/java/datastore/queries#query_interface_example

`Key key = keyFactory.newKey(idValue);` then use it inside Entity.newBuilder(), so that we use the idValue as the key.

Then we simply check if it exists when we create the newUser entity, check example 8: https://www.programcreek.com/java-api-examples/?class=com.google.cloud.datastore.Entity&method=newBuilder

Make these changes in the next PR.

Reply:
If idValue isn't unique, maybe just use the email? Should be good enough.

Reply:
Hi David!

Sorry for the late, I have already solved that issue, and I am currently working on constructing a well-established user system. I think I will submit another PR before today's meeting.